syntax = "proto3";
// Namespace
package greetings;

service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
    rpc GetVector (NoParams) returns ( stream VectorElement ) {}
}

message HelloRequest {
    string name = 1;
}

message HelloReply {
    string message = 1;
}

message NoParams {}

message VectorElement {
    int32 num = 1;
}